src/transformStateWithClones.js

```diff
@@ -1,13 +1,33 @@
-'use strict';
-
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
+"use strict";
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [{ ...state }];
+
+  for (let i = 0; i < actions.length; i++) {
+    const valueBeforePassing =
+      i === 0 ? { ...resultArray[i] } : { ...resultArray[i - 1] };
+
+    switch (actions[i].type) {
+      case "addProperties":
+        const addProperties = actions[i].extraData;
+
+        resultArray[i] = Object.assign(valueBeforePassing, addProperties);
```

Comment:
Instead of using Object.assign to add properties, you can simply use the spread operator to add properties to an object. This is a more modern and cleaner approach.